menas/src/main/scala/za/co/absa/enceladus/menas/repositories/OozieRepository.scala

```diff
@@ -0,0 +1,338 @@
+
+  @Value("${za.co.absa.enceladus.menas.oozie.mavenConformanceJarLocation:}")
+  val conformanceJarPath: String = ""
+
+  @Value("${za.co.absa.enceladus.menas.oozie.mavenRepoLocation:}")
+  val mavenRepoLocation: String = ""
+
+  @Value("${za.co.absa.enceladus.menas.oozie.menasApiURL:}")
+  val menasApiURL: String = ""
+
+  @Value("${za.co.absa.enceladus.menas.oozie.splineMongoURL:}")
+  val splineMongoURL: String = ""
+
+  val classLoader = Thread.currentThread().getContextClassLoader()
+  val workflowTemplate = getTemplateFile("scheduling/oozie/workflow_template.xml")
+  val coordinatorTemplate = getTemplateFile("scheduling/oozie/coordinator_template.xml")
+  val namenode = hadoopConf.get("fs.defaultFS")
+  val resourceManager = hadoopConf.get("yarn.resourcemanager.address")
+  val dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'")
+  val logger = LoggerFactory.getLogger(this.getClass)
+
+  override def afterPropertiesSet() {
+    logger.info(s"Enceladus Jar Location: $enceladusJarLocation")
+    logger.info(s"Stanrdardization Jar Path: $standardizationJarPath")
+    logger.info(s"Conformance Jar Path: $conformanceJarPath")
+    //ensure that relevant jars are properly loaded in HDFS, otherwise initialize
+    this.initializeJars()
+  }
+
+  private def validateProperties(): Boolean = {
+    Seq((oozieScheduleHDFSPath, "za.co.absa.enceladus.menas.oozie.schedule.hdfs.path"),
+      (enceladusJarLocation, "zza.co.absa.enceladus.menas.oozie.enceladusJarLocation"),
+      (standardizationJarPath, "za.co.absa.enceladus.menas.oozie.mavenStandardizationJarLocation"),
+      (conformanceJarPath, "za.co.absa.enceladus.menas.oozie.mavenConformanceJarLocation"),
+      (mavenRepoLocation, "za.co.absa.enceladus.menas.oozie.mavenRepoLocation"),
+      (menasApiURL, "za.co.absa.enceladus.menas.oozie.menasApiURL"),
+      (splineMongoURL, "za.co.absa.enceladus.menas.oozie.splineMongoURL")).map(p => validateProperty(p._1, p._2)).reduce(_ && _)
+  }
+
+  private def validateProperty(prop: String, propName: String): Boolean = {
+    if (prop == null || prop.isEmpty) {
+      logger.warn(s"Oozie support disabled. Missing required configuration property $propName")
+      false
+    }
+    true
+  }
+
+  private def initializeJars() {
+    if (this.isOozieEnabled) {
+      val hdfsStdPath = new Path(s"$enceladusJarLocation$standardizationJarPath")
+      val hdfsConfPath = new Path(s"$enceladusJarLocation$conformanceJarPath")
+      val mavenStdPath = s"$mavenRepoLocation$standardizationJarPath"
+      val mavenConfPath = s"$mavenRepoLocation$conformanceJarPath"
+
+      if (!hadoopFS.exists(hdfsStdPath) || hadoopFS.getStatus(hdfsStdPath).getCapacity == 0) {
+        logger.info(s"Uploading standardization jar from $mavenStdPath to $hdfsStdPath")
+        val resFuture = this.downloadFile(mavenStdPath, hdfsStdPath)
+        resFuture.onSuccess({ case (u: Unit) => logger.info(s"Standardization jar loaded to $hdfsStdPath") })
```

Comment:
```suggestion
        resFuture.onSuccess{ case _ => logger.info(s"Standardization jar loaded to $hdfsStdPath") }
```
Unused variables, extra parenthesis